File: examples/threads/MaxThreads.java

/**
 * See how many threads can be created, or, how big a quagmire we can create....
 * 
 * @author amit
 */
public class MaxThreads
{
    final static int MAX = 50000;

    public static void main(String args[]) throws InterruptedException {
        for (int i = 0; i < MAX; i++) {
            Integer I = i;
            new nuts(I.toString()).start();
        }
    }
}

class nuts extends Thread
{
    public nuts(String s) {
        super(s);
    }


    public void run() {
        System.out.println("Thread number " + this.getName());
        /* System.out.flush(); */
        try {
            Thread.sleep(20000); // in millisecs
        } catch (InterruptedException e) {
            System.err.println(e);
        }
    }
}

File: examples/golang/threads/MaxThreads.go
//Author: Amit Jain & Paul Ellis
package main

import (
	"fmt"
	"sync"
	"time"
)

const MAX = 1000000

func run(i int, wg *sync.WaitGroup){
	fmt.Println("Thread Number: ", i)
	time.Sleep(time.Second*20)
	wg.Done()
}

func main(){

	var wg sync.WaitGroup

	//Start as many go routines as we can
	for i := 0; i< MAX; i++{
		wg.Add(1)
		go run(i, &wg)
	}

	wg.Wait()

}
